File: src/python/hollow_inverted_pyramid_pattern.py
# Python program to print a hollow 
# inverted pyramid pattern 
''' 
O/P

***********                                                                                                                                   
 *       *                                                                                                                                    
  *     *                                                                                                                                     
   *   *                                                                                                                                      
    * *                                                                                                                                       
     *   '''



def printPattern(n) : 
	
	for i in range(1,n+1) : 
		
		# Print spaces 
		for j in range(1,i) : 
			print (" ", end="") 
			
		# Print hollow inverted pyramid 
		for j in range(1,(n * 2 - (2 * i - 1)) 
									+1) : 
			if (i == 1 or j == 1 or
			j == (n * 2 - (2 * i - 1))) : 
				print ("*", end="") 
			else : 
				print(" ", end="") 
				
		# next line 
		print (""), 

# Driver code 
n = 6
printPattern(n) 

File: src/C++/hollow_inverted_pyramid_pattern.cpp
// CPP program to print a hollow 
// inverted pyramid pattern 
/*
o/p

***********
 *       *
  *     *
   *   *
    * *
     *
*/  

#include <iostream> 
using namespace std; 
void printPattern(int); 
int main() 
{ 
	int n = 6; 

	printPattern(n); 
} 
void printPattern(int n) 
{ 
	int i, j; 
	for (i = 1; i <= n; i++) { 
		// Print spaces 
		for (j = 1; j < i; j++) { 
			cout << " "; 
		} 

		// Print hollow inverted pyramid 
		for (j = 1; j <= (n * 2 - (2 * i - 1)); j++) { 

			if (i == 1 || j == 1 || j == (n * 2 - (2 * i - 1))) { 
				cout << "*"; 
			} else { 
				cout << " "; 
			} 
		} 

		// next line 
		cout << endl; 
	} 
} 
